File: avalon-modules/avalon-authorization/src/test/java/io/github/lishangbu/avalon/authorization/TestEnvironmentApplication.java
package io.github.lishangbu.avalon.authorization;

import org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase;
import org.springframework.boot.persistence.autoconfigure.EntityScan;
import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.springframework.context.annotation.Bean;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.testcontainers.postgresql.PostgreSQLContainer;
import org.testcontainers.utility.DockerImageName;

/// 测试环境自动配置类
///
/// @author lishangbu
/// @since 2025/8/20
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
@EntityScan(basePackages = "io.github.lishangbu.avalon.authorization.entity")
@EnableJpaRepositories(basePackages = "io.github.lishangbu.avalon.authorization.repository")
public class TestEnvironmentApplication {

    /// PostgreSQL 测试容器 Bean
    ///
    /// 使用 @ServiceConnection 注解，Spring Boot 会自动配置数据源连接
    ///
    /// @return PostgreSQL 容器实例
    @Bean
    @ServiceConnection
    PostgreSQLContainer postgresContainer() {
        return new PostgreSQLContainer(DockerImageName.parse("postgres"));
    }
}

File: avalon-modules/avalon-authorization/src/test/kotlin/io/github/lishangbu/avalon/authorization/TestEnvironmentApplication.kt
package io.github.lishangbu.avalon.authorization

import io.github.lishangbu.avalon.authorization.service.impl.DefaultOAuth2AuthorizationConsentService
import io.github.lishangbu.avalon.authorization.service.impl.DefaultOAuth2AuthorizationService
import io.github.lishangbu.avalon.authorization.service.impl.DefaultRegisteredClientRepository
import org.babyfish.jimmer.spring.repository.EnableJimmerRepositories
import org.springframework.boot.SpringBootConfiguration
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Import
import org.testcontainers.postgresql.PostgreSQLContainer
import org.testcontainers.utility.DockerImageName

/**
 * 授权模块测试环境配置
 *
 * 为仓储测试提供最小化的 Spring Boot 上下文
 */
@SpringBootConfiguration
@EnableAutoConfiguration
@EnableJimmerRepositories("io.github.lishangbu.avalon.authorization.repository")
@ComponentScan(basePackages = ["io.github.lishangbu.avalon.authorization.repository"])
@Import(
    DefaultRegisteredClientRepository::class,
    DefaultOAuth2AuthorizationService::class,
    DefaultOAuth2AuthorizationConsentService::class,
)
class TestEnvironmentApplication {
    /** 创建 PostgreSQL 测试容器 */
    @Bean
    @ServiceConnection
    fun postgresContainer(): PostgreSQLContainer = PostgreSQLContainer(DockerImageName.parse("postgres"))
}
